### PR TITLE
Add support for rails 7.1

### LINKED ISSUE #215: ColumnPresenceChecker is incompatible with Rails' new default `belongs_to` validation behavior
Starting in the next major release of Rails, thanks to rails/rails#46522, the validation behavior of `belongs_to` has changed to include an `:if`, to only validate if the underlying foreign key attribute has changed. This does not play well with the way `ColumnPresenceChecker` analyzes whether a particular column is consistent with its AR model:

https://github.com/djezzzl/database_consistency/blob/18e4aec9bb33f71f7de1d61ee906774a10c987b3/lib/database_consistency/checkers/validators_fraction_checkers/column_presence_checker.rb#L56-L68
https://github.com/djezzzl/database_consistency/blob/18e4aec9bb33f71f7de1d61ee906774a10c987b3/lib/database_consistency/checkers/validators_fraction_checkers/column_presence_checker.rb#L36-L38
https://github.com/djezzzl/database_consistency/blob/18e4aec9bb33f71f7de1d61ee906774a10c987b3/lib/database_consistency/checkers/validators_fraction_checkers/column_presence_checker.rb#L7

I'm not sure off-hand what the ideal change is here to make this continue to work, but wanted to raise the issue now while the next major release of Rails is still in active development!

### PR DESCRIPTION
**Pull request by chaadow**
fixes #215 

rails 7.1 recently added  this new default [configuration option](https://guides.rubyonrails.org/configuring.html#config-active-record-belongs-to-required-validates-foreign-key )

When set to false ( default in rails 7.1 ) an `if` option is added to the belongs_to option. which makes it not compatible with the current `WEAK_OPTIONS` array.

To cover this new case, and as a pre-check, we make sure that we're on rails 7.1.0 minimum as well as the ActiveRecord option ( `ActiveRecord.belongs_to_required_validates_foreign_key` ) is set to false to mitigate issues with real `if` conditions. We also check that the if condition is coupled with a `message` option set to `required`

This commit also adds rails 7.1 to the CI matrix.

cc @djezzzl @agrobbin

### CODE REVIEW

**Comment by chaadow** [comment]
@djezzzl btw I'm going to go ahead and try to add rails 7.1 to the github CI matrix. as well as add a unit test to the presence checker 👍🏼

File: lib/database_consistency/checkers/validators_fraction_checkers/column_presence_checker.rb [hunk 1]
```diff
@@ -34,7 +34,23 @@ def report_template(status, column_name:, error_slug: nil)
       end
 
       def weak_option?
```
Comment (chaadow) [inline lib/database_consistency/checkers/validators_fraction_checkers/column_presence_checker.rb:36]: @djezzzl after refactoring, and considering the edge case you mentioned in the issue, ( and thanks to specs for helping me out), I figured all cases can be covered by simply tweaking the `weak_option?` to make it return false if one of the validators has the `if` condition + `message: :required`  **only when rails 7.1 + the new config is set to false**. 👍🏼

**Comment by chaadow** [comment]
rebased and squashed

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -51,7 +51,7 @@ jobs:
           - ruby-version: '3.1'
             gemfile: 'gemfiles/ar_6_1.gemfile'
           - ruby-version: '3.2'
-            gemfile: 'gemfiles/ar_7_0.gemfile'
+            gemfile: ['gemfiles/ar_7_0.gemfile', 'gemfiles/ar_7_1.gemfile']
```
Comment (djezzzl) [inline .github/workflows/tests.yml:54]: Sorry, this shouldn't be an array. I will split it.

**Comment by djezzzl** [comment]
Thank you very much for your contribution!

I'll update a changelog and release it today! Please let me know if you have any more ideas to make this gem more helpful. 

Have a great day and the upcoming week!

**Pull request merged**